exa_error.go

```diff
@@ -0,0 +1,75 @@
+package error_reporting_go
+
+import (
+	"fmt"
+	"reflect"
+	"strings"
+)
+
+type ErrorMessageBuilder struct {
+	errorCode  string
+	message    string
+	parameters [][]string
+	mitigation string
+}
+
+func ExaError(errorCode string) *ErrorMessageBuilder {
+	return &ErrorMessageBuilder{
+		errorCode: errorCode,
+	}
+}
+
+func (builder *ErrorMessageBuilder) Message(message string) *ErrorMessageBuilder {
+	builder.message = message
+	return builder
+}
+
+func (builder *ErrorMessageBuilder) Parameter(arguments ...interface{}) *ErrorMessageBuilder {
+	validateArguments(arguments)
+	builder.parameters = append([][]string{{fmt.Sprintf("%v", arguments[0]), fmt.Sprintf("%v", arguments[1])}})
+	return builder
+}
+
+func (builder *ErrorMessageBuilder) Mitigation(mitigation string) *ErrorMessageBuilder {
+	builder.mitigation = mitigation
+	return builder
+}
+
+func validateArguments(arguments []interface{}) {
```

Comment:
I don't like that method. These checks should be done at compile time, not at runtime.